contributors/design-proposals/storage/container-storage-interface-inline-volumes.md

```diff
@@ -0,0 +1,148 @@
+# In-line CSI volumes in Pods
+
+Author: @jsafrane
+
+## Goal
+* Define API and high level design for in-line CSI volumes in Pod
+
+## Motivation
+Currently, CSI can be used only though PersistentVolume object. All other persistent volume sources support in-line volumes in Pods, CSI should be no exception. There are two main drivers:
+* We want to move away from in-tree volume plugins to CSI, as designed in a separate proposal https://github.com/kubernetes/community/pull/2199/. In-line volumes should use CSI too.
+* CSI drivers can be used to provide Secrets-like volumes to pods, e.g. providing secrets from a remote vault. We don't want to force users to create PVs for each secret, we should allow to use them in-line in pods as regular Secrets or Secrets-like Flex volumes.
+
```

Comment:
You hint at it a bit, but might be good to explicitly add it as a motivation:
* Reach feature parity with FlexVolume api and allow for its deprecation.

Reply:
Added a new motivation item.